diff --git a/src/system/zephyr/network.c b/src/system/zephyr/network.c
index 731da21..468f8c1 100644
--- a/src/system/zephyr/network.c
+++ b/src/system/zephyr/network.c
@@ -52,7 +52,7 @@ z_result_t _z_socket_set_non_blocking(const _z_sys_net_socket_t *sock) {
 
 z_result_t _z_socket_accept(const _z_sys_net_socket_t *sock_in, _z_sys_net_socket_t *sock_out) {
     struct sockaddr naddr;
-    unsigned int nlen = sizeof(naddr);
+    socklen_t nlen = sizeof(naddr);
     sock_out->_fd = -1;
     int con_socket = accept(sock_in->_fd, &naddr, &nlen);
     if (con_socket < 0) {
@@ -373,7 +373,7 @@ void _z_close_udp_unicast(_z_sys_net_socket_t *sock) {
 
 size_t _z_read_udp_unicast(const _z_sys_net_socket_t sock, uint8_t *ptr, size_t len) {
     struct sockaddr_storage raddr;
-    unsigned int addrlen = sizeof(struct sockaddr_storage);
+    socklen_t addrlen = sizeof(struct sockaddr_storage);
 
     ssize_t rb = recvfrom(sock._fd, ptr, len, 0, (struct sockaddr *)&raddr, &addrlen);
     if (rb < (ssize_t)0) {
@@ -413,7 +413,7 @@ z_result_t _z_open_udp_multicast(_z_sys_net_socket_t *sock, const _z_sys_net_end
     z_result_t ret = _Z_RES_OK;
 
     struct sockaddr *lsockaddr = NULL;
-    unsigned int addrlen = 0;
+    socklen_t addrlen = 0;
     if (rep._iptcp->ai_family == AF_INET) {
         lsockaddr = (struct sockaddr *)z_malloc(sizeof(struct sockaddr_in));
         if (lsockaddr != NULL) {
@@ -516,7 +516,7 @@ z_result_t _z_listen_udp_multicast(_z_sys_net_socket_t *sock, const _z_sys_net_e
     z_result_t ret = _Z_RES_OK;
 
     struct sockaddr *lsockaddr = NULL;
-    unsigned int addrlen = 0;
+    socklen_t addrlen = 0;
     if (rep._iptcp->ai_family == AF_INET) {
         lsockaddr = (struct sockaddr *)z_malloc(sizeof(struct sockaddr_in));
         if (lsockaddr != NULL) {
@@ -683,7 +683,7 @@ void _z_close_udp_multicast(_z_sys_net_socket_t *sockrecv, _z_sys_net_socket_t *
 size_t _z_read_udp_multicast(const _z_sys_net_socket_t sock, uint8_t *ptr, size_t len, const _z_sys_net_endpoint_t lep,
                              _z_slice_t *addr) {
     struct sockaddr_storage raddr;
-    unsigned int raddrlen = sizeof(struct sockaddr_storage);
+    socklen_t raddrlen = sizeof(struct sockaddr_storage);
 
     ssize_t rb = 0;
     do {
